editor/gui/editor_scroll_box.cpp

```diff
@@ -0,0 +1,298 @@
+		second_button->set_disabled(scroll_bar->get_value() == 0);
+	} else {
+		first_button->set_disabled(scroll_bar->get_value() == 0);
+		second_button->set_disabled(scroll_bar->get_value() + scroll_bar->get_page() == scroll_bar->get_max());
+	}
+}
+
+bool EditorScrollBox::_set(const StringName &p_name, const Variant &p_value) {
+	if (p_name == "vertical") {
+		if ((bool)p_value == is_vertical()) {
+			return true;
+		}
+		set_vertical((bool)p_value);
+		if (is_vertical() != (bool)p_value) {
+			return true;
+		}
+
+		// Switch the size flags.
+		if (is_vertical()) {
+			set_v_size_flags(get_h_size_flags());
+			set_h_size_flags(SIZE_FILL);
+		} else {
+			set_h_size_flags(get_v_size_flags());
+			set_v_size_flags(SIZE_FILL);
+		}
+
+		_update_buttons_icon_and_tooltip();
+		_update_buttons();
+		_update_scroll_container();
+
+		emit_signal(SNAME("vertical_changed"), p_value);
+		return true;
+	}
+	return false;
+}
+
+void EditorScrollBox::_validate_property(PropertyInfo &p_property) const {
+	if (is_fixed && p_property.name == "vertical") {
+		p_property.usage = PROPERTY_USAGE_NONE;
+	}
+}
+
+// Scrolls down or right based on the class layout (vertical or horizontal).
+void EditorScrollBox::_accessibility_action_scroll_forward(const Variant &p_data) {
+	_scroll(true);
+}
+
+// Scrolls up or left based on the class layout (vertical or horizontal).
+void EditorScrollBox::_accessibility_action_scroll_backward(const Variant &p_data) {
+	_scroll(false);
+}
+
+void EditorScrollBox::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ACCESSIBILITY_UPDATE: {
+			RID ae = get_accessibility_element();
+			ERR_FAIL_COND(ae.is_null());
+			DisplayServer::get_singleton()->accessibility_update_set_role(ae, DisplayServer::AccessibilityRole::ROLE_SCROLL_VIEW);
+			if (is_vertical()) {
+				DisplayServer::get_singleton()->accessibility_update_add_action(ae, DisplayServer::AccessibilityAction::ACTION_SCROLL_UP, callable_mp(this, &EditorScrollBox::_accessibility_action_scroll_backward));
+				DisplayServer::get_singleton()->accessibility_update_add_action(ae, DisplayServer::AccessibilityAction::ACTION_SCROLL_DOWN, callable_mp(this, &EditorScrollBox::_accessibility_action_scroll_forward));
+			} else {
+				DisplayServer::get_singleton()->accessibility_update_add_action(ae, DisplayServer::AccessibilityAction::ACTION_SCROLL_LEFT, callable_mp(this, &EditorScrollBox::_accessibility_action_scroll_backward));
+				DisplayServer::get_singleton()->accessibility_update_add_action(ae, DisplayServer::AccessibilityAction::ACTION_SCROLL_RIGHT, callable_mp(this, &EditorScrollBox::_accessibility_action_scroll_forward));
+			}
+		} break;
+
+		case NOTIFICATION_TRANSLATION_CHANGED:
+		case NOTIFICATION_LAYOUT_DIRECTION_CHANGED: {
+			_update_scroll_container();
+		}
+			[[fallthrough]];
+
+		case NOTIFICATION_ENTER_TREE:
+		case NOTIFICATION_THEME_CHANGED: {
+			_update_buttons_icon_and_tooltip();
+		}
+			[[fallthrough]];
```

Comment:
This doesn't make much sense. You can just call `_update_buttons()`, it's a single line either way.